user/user_lego_sensor.h

```diff
@@ -16,9 +16,18 @@
 #include <lego.h>
 #include <lego_sensor_class.h>
 
+/* The USER_LEGO_SENSOR_TEXT_VALUE_SIZE can be a maximum of PAGE_SIZE-2
+ * to take into account a trailing newline (added later) and NULL. We are
+ * limiting it to much less to save memory for each user_lego_sensor 
+ * instance.
+ */
+
+#define USER_LEGO_SENSOR_TEXT_VALUE_SIZE (512-2)
```

Comment:
Which is exactly why I mentioned it - but here's the thing - I can't easily tell if there is supposed to be a triling newline because the user wanted it, or if they forgot to use the `-e` option - I think the better thing to do is to NOT add the `\n` when we copy the string to the buffer for the read operation.